### DISCUSSION TITLE
Trouble filtering entries by taxonomy in GraphQL

### DISCUSSION THREAD

**Discussion by andyra** [statamic/cms#4983] [Troubleshooting] [answered]
I'm having trouble filtering entries by a taxonomy/term using GraphQL. Putting together the docs on [Taxonomy conditions](https://statamic.dev/conditions#taxonomy-conditions) and [GraphQL filtering](https://statamic.dev/graphql#filtering), it looks like I should be able to filter entries like this:

```
{
  entries(
    collection: "wikis"
    filter: {
      taxonomy: { tags: "buildings" }
    }
  ) {
    data {
      title
    }
  }
}
```

Instead, I get back a list of _all_ the entries in the "wikis" collection instead of just the ones with the specific term I requested. Is this a bug with the GraphQL API, or am I building the query wrong? I'm not getting any errors or anything.

I'm on Statamic **3.2.23**

**Comment by sebszocinski** [comment, marked as answer]
hey @andyra try this:
`taxonomy: { contains: "buildings" }`

**Reply by andyra** [reply]
Worked like a charm!

**Reply by grischaerbe** [reply]
How would you query entries that contain one of many taxonomy terms? There's an [unanswered forum post](https://statamic.com/forum/5577-filter-entries-by-multiple-taxonomies-via-graphql) about this too.
Let's say there's a taxonomy "tags" with the taxonomy terms "food" and "health" and two posts of a blog: post A has tag "food", post B has tag "health". Querying for both posts by providing both taxonomy terms isn't possible at the moment. This is how it should look like if following the taxonomy filter of the antlers tag:

```graphql
{
  entries(
    collection: "posts"
    filter: {
      taxonomy: { tags: "food|health" }
    }
  ) {
    data {
      title
    }
  }
}
```

As stated above this is returning every post of the collection. The docs note that this should be possible with the Antlers tag, I've never used them so I can't tell.

@sebszocinski approach would be like this:

```graphql
{
  entries(
    collection: "posts"
    filter: {
      tags: { in: "food|health" } // the filter `in` also is not explicit about being `AND` or `OR`
    }
  ) {
    data {
      title
    }
  }
}
```

This returns no entries at all. So I would actually not mark this question as answered. This is a common use case for taxonomies and something that seems unreflected in the GraphQL API.

**Comment by wnzirngibl** [comment]
> How would you query entries that contain one of many taxonomy terms? 

On Statamic 4.36.0 I still have the exact same question: how would you write a query for entries having ONE of multiple taxonomy terms? I searched the documentation, and I tried lots of different queries, but I can't find out how to send an OR query by graphQL in Statamic.

I managed to send an AND query though - this query successfully returns all entries having term "social-media" AND term "screen-design":
```
entries(
  filter: {
    categories_content_campaigns: [
      {contains: "categories::social-media"},
      {contains: "categories::screen-design"}
    ]
  },
  collection: "blog"
)
```
But I can't manage to build an OR query - returning all entries having at least one of the terms.

The documentation says, I can use any of the supported tag conditions:
https://statamic.dev/graphql#using-filters
linking to: https://statamic.dev/conditions

But especially the taxonomy conditions cannot really be used in graphQL: https://statamic.dev/conditions#taxonomy-conditions
At least I couldn't find out how I would write a tag condition like this in graphQL:
`taxonomy:tags:any="social-media|screen-design"`

I found a similar issue, without any answers unfortunately: https://github.com/statamic/ideas/issues/914

Does anybody have an idea how to build a simple OR query like that in graphQL?
